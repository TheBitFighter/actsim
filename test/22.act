defproc test1 (bool a, b)
{
  bool q;
  chp {
     b-;
     q+;q+;q+;
     log("wait-a");
     [a]; 
     log("set-b");
     q+;q+;q+;
     log ("pre-done");
     b+;
     log ("done-1")
  }
}

defproc test2 (bool p, q)
{
   chp {
     log("set p");
     p+;
     log ("wait q");
     [q];
     log ("done-2")
   }
}

defproc test ()
{
   test1 x;
   test2 y;
   x.a = y.p;
   x.b = y.q;
}

test t;
