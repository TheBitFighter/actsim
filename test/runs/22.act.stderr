WARNING: test2<>: substituting chp model (requested prs, not found)
WARNING: test1<>: substituting chp model (requested prs, not found)
